sqlglot/parser.py

```diff
@@ -4090,16 +4090,15 @@ def _parse_interval(self, match_interval: bool = True) -> t.Optional[exp.Add | e
         if this and this.is_number:
             this = exp.Literal.string(this.name)
         elif this and this.is_string:
-            parts = this.name.split()
-
-            if len(parts) == 2:
+            parts = exp.INTERVAL_STRING_RE.findall(this.name)
+            if len(parts) == 1:
                 if unit:
-                    # This is not actually a unit, it's something else (e.g. a "window side")
+                    # Unconsume the eagerly-parsed unit, since the real unit was part of the string
                     unit = None
                     self._retreat(self._index - 1)
 
-                this = exp.Literal.string(parts[0])
-                unit = self.expression(exp.Var, this=parts[1].upper())
+                this = exp.Literal.string(parts[0][0])
+                unit = self.expression(exp.Var, this=parts[0][1].upper())
```

Comment:
Do we have to set `unit = None`? Seems like we'll set it to `exp.Var` right after the branch anyways